### DISCUSSION TITLE
Filter realtime updates with nested relations

### DISCUSSION THREAD

**Discussion by akselw** [supabase/supabase#1206] [Questions] [answered]
Is there a way to filter realtime updates with the JavaScript library using nested filters (if that's the correct term)? Something like this:

```
const table_a = supabase
  .from('table_a:table_b.table_c_id=eq.someValue')
  .on('*', payload => {
    console.log('Change received!', payload)
  })
  .subscribe()
```

Where `table_a` has a column `table_b_id` with a foreign key relationship to `table_b`, and `table_b` has a column `table_c_id` with a foreign key relationship to `table_c` (not sure if that last relationship would actually be required).

I haven't seen it documented anywhere, and it hasn't worked trying out different syntaxes that seemed plausible.

**Comment by silentworks** [comment, marked as answer]
Would it be easier to listen to realtime updates on the resulting table first and then work out if a change was made? It's not very clear as to why you would want to listen through a relationship.

**Reply by akselw** [reply]
But it _is_ actually `table_a` I want to listen to changes to, not `table_b` or `table_c`. I only want to filter out changes to rows in `table_a` that don't have a relationship to a particular row in `table_c`.

I could achieve the same thing by listening to all changes to `table_a`, and then check each time to see whether the row has the correct relationship, but if there is a way to just specify that to the JavaScript-library, then that would be preferable.

**Reply by akselw** [reply]
So specifically, I have something like this: I'm making something like a chat app, where you can have a conversation (table `conversation`), which has messages (table `message`), and messages can have replies (table `reply`).

Table `conversation` has a column `id`, table `message` has a column `conversation_id`, and table `reply` has a column `message_id`.

What I want is to subscribe to all changes to both messages and replies for a specific conversation. I already have the following working to subscribe to messages:

```
supabase
    .from(`message:conversation_id=eq.${conversationId}`)
    .on('*', payload => {
        handlePayload(payload);
    })
    .subscribe();
```

And what I want in addition to that is some way to subscribe to all the changes to replies, that are in the conversation. Which means replies where the message the reply is connected to has a specific conversation_id. Is that possible?

**Reply by silentworks** [reply]
A way you could design this to be able to easily do realtime on the replies based on the conversation is to add a `conversation_id` to the reply table. It would make the lookup easier. Otherwise, you could start a new subscription on all replies based on the `message.id` that gets returned in the payload: 

```js
supabase
    .from(`message:conversation_id=eq.${conversationId}`)
    .on('*', payload => {
        handlePayload(payload);
        supabase.from(`reply:message_id=eq.${payload.id}`)
            .on('*', payload => { handleReplyPayload(payload) })
            .subscribe();
    })
    .subscribe();
```

Although this might work (I haven't tested this code) I would rather update the table schema and be more efficient with the lookup. Also if you should do it this way, remember to assign each subscription to a variable so you can unsubscribe later or it could lead to a memory leak in your application.

**Reply by akselw** [reply]
Okay, yeah, then I'll do that. Thanks for the help! Do you know if there is a way to place a constraint on reply.conversation_id in postgres, so that I reply.conversation_id always matches the conversation_id of the message in reply.message_id?

**Reply by silentworks** [reply]
You can create a stored procedure that could do the lookup for you and add it based on a trigger. If I get some time I will try and write one tomorrow to show how you would do this.

**Reply by KingMCMS** [reply]
Was soll das

**Reply by elhe26** [reply]
Hey @silentworks, did you do it?
